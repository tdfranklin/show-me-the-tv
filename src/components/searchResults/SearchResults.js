import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { GridList, GridTile } from 'material-ui/GridList';
import IconButton from 'material-ui/IconButton';
import ZoomIn from 'material-ui/svg-icons/action/zoom-in';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import axios from 'axios';

class TVShowResults extends Component {
    constructor(props) {
        super(props);
        this.state = {
            open: false,
            currentItem: ''
        };
    };

    handleOpen = id => {
        const apiUrl = this.props.content === 'tv' ? '/api/tv' : '/api/movie';
        axios.get(`${apiUrl}/${id}`)
        .then(res => {
            this.setState({open: true, currentItem: res.data});
        })
        .catch(err => console.log(err));
    }

    handleClose = () => {
        this.setState({ open: false });
    }

    render() {
        let itemContent;
        const { display, content } = this.props;
        let dialogDisplay;

        if(content === 'tv') {
            dialogDisplay = (
                <div>
                    <img src={this.state.currentItem.image} alt="" style={{ width: '30%', height: '50%', float: 'left', marginRight: 50 }} />
                    <a href={this.state.currentItem.website} target='_blank'><h1 style={{ paddingTop: 100, textDecoration: 'underline', color: 'white' }}>{this.state.currentItem.name}</h1></a>
                    <br />
                    <p>{this.state.currentItem.summary}</p>
                    <br />
                    <p>Genre: {this.state.currentItem.genres}</p>
                    <p>Episodes: {this.state.currentItem.episodes}</p>
                    <p>Seasons: {this.state.currentItem.seasons}</p>                    
                    <p>Rating: {this.state.currentItem.rating}</p>
                    <p>Status: {this.state.currentItem.status}</p>
                </div>
            )
        } else {
            dialogDisplay = (
                <div>
                    <img src={this.state.currentItem.image} alt="" style={{ width: '30%', height: '50%', float: 'left', marginRight: 50 }} />
                    <a href={this.state.currentItem.website} target='_blank'><h1 style={{ paddingTop: 100, textDecoration: 'underline', color: 'white' }}>{this.state.currentItem.name}</h1></a>
                    <br />
                    <p>{this.state.currentItem.summary}</p>
                    <br />
                    <p>Genre: {this.state.currentItem.genres}</p>
                    <p>Runtime: {this.state.currentItem.runtime}</p>
                    <p>Release Date: {this.state.currentItem.release}</p>
                    <p>Rating: {this.state.currentItem.rating}</p>
                </div>
            )
        }

        if(display) {
            itemContent = (
                <GridList cols={3}>
                    {display.map(item => (
                        <GridTile
                            title={item.name}
                            key={item.id}
                            actionIcon={
                                <IconButton onClick={() => this.handleOpen(item.id)}>
                                    <ZoomIn color="white" />
                                </IconButton>
                            }
                        >
                            <img src={item.image} alt="" />
                        </GridTile>
                    ))}
                </GridList>
            );
        } else {
            itemContent = null;
        }

        const actions = [
            <FlatButton label="Close" primary={true} onClick={this.handleClose} />
        ]

        const dialogStyle = {
            width: '100%',
            maxWidth: '90%',
            marginTop: '-8%',
        }
        return (
            <div>
                {itemContent}
                <Dialog 
                    actions={actions}
                    modal={false}
                    open={this.state.open}
                    onRequestClose={this.handleClose}
                    contentStyle={dialogStyle}
                    autoScrollBodyContent={true}
                >
                    {dialogDisplay}
                </Dialog>
            </div>
        );
    }
}

TVShowResults.propTypes = {
    display: PropTypes.array.isRequired,
    content: PropTypes.string.isRequired
}

export default TVShowResults;