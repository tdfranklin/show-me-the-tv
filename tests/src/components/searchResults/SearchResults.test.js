import React from 'react';
import ReactDOM from 'react-dom';
import SearchResults from '../../../../src/components/searchResults/SearchResults';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { GridList, GridTile } from 'material-ui/GridList';
import IconButton from 'material-ui/IconButton';
import ZoomIn from 'material-ui/svg-icons/action/zoom-in';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';

const apiKey = 'fb6a1d3f38c3d97f67df6d141f936f29';
const apiSearchUrl = 'https://api.themoviedb.org/3';
const apiImageUrl = 'https://image.tmdb.org/t/p/original';
const axios = require('axios');
const popularShows = [];

configure({ adapter: new Adapter() });

beforeAll(() => {
    axios
        .get(`${apiSearchUrl}/tv/popular?api_key=${apiKey}`)
        .then(payload => {
            payload.data.results.forEach(show => {
                popularShows.push({
                    id: show.id,
                    name: show.name,
                    image: `${apiImageUrl}${show.backdrop_path}`
                });
            });
        })
        .catch(err => console.log(err));
});

describe('SearchResults', () => {
    it('matches the snapshot', () => {
        const tree = shallow(<SearchResults display={[]} content='tv' />);
        expect(toJson(tree)).toMatchSnapshot();
    });

    it('accepts display and content prop', () => {
        const wrapper = shallow(<SearchResults display={popularShows} content='tv' />);
        expect(wrapper.instance().props.display).toBe(popularShows);
        expect(wrapper.instance().props.content).toBe('tv');
    });

    it('changes data based on display array', () => {
        const wrapper = shallow(<SearchResults display={[]} content='tv' />);
        const data = wrapper.get(0);
        wrapper.setProps({display: popularShows});
        expect(wrapper.get(0)).not.toBe(data);
    });

    it('changes data based on content string', () => {
        const wrapper = shallow(<SearchResults display={[]} content='tv' />);
        let data = wrapper.get(0);
        wrapper.setProps({content: 'movies'});
        expect(wrapper.get(0)).not.toBe(data);
    });

    describe('Material UI Components', () => {
        const wrapper = shallow(<SearchResults display={popularShows} content='tv' />);

        describe('GridList', () => {
            let node = wrapper.find(GridList);
            
            it('renders component', () => {
                expect(node).toBeTruthy();
            });

            it('has 3 columns', () => {
                expect(node.props().cols).toBe(3);
            });
        });

        describe('GridTile', () => {
            const node = wrapper.find(GridTile);
            
            it('renders component', () => {
                expect(node).toBeTruthy();
            });
        });

        describe('IconButton', () => {
            const node = wrapper.find(IconButton);
            
            it('renders component', () => {
                expect(node).toBeTruthy();
            });
        });

        describe('ZoomIn', () => {
            const node = wrapper.find(ZoomIn);
            
            it('renders component', () => {
                expect(node).toBeTruthy();
            });
        });

        describe('Dialog', () => {
            const node = wrapper.find(Dialog);
            
            it('does not render component', () => {
                expect(node).toBeTruthy();
            });
        });

        describe('FlatButton', () => {
            const node = wrapper.find(FlatButton);
            
            it('does not render component', () => {
                expect(node).toBeTruthy();
            });
        });
    });
});