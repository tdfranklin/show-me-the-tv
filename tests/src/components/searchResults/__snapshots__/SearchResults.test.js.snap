// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults matches the snapshot 1`] = `
<div>
  <GridList
    cellHeight={180}
    cols={3}
    padding={4}
  />
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Close"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
        />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={true}
    contentStyle={
      Object {
        "marginTop": "-8%",
        "maxWidth": "90%",
        "width": "100%",
      }
    }
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
  >
    <div>
      <img
        alt=""
        style={
          Object {
            "float": "left",
            "height": "50%",
            "marginRight": 50,
            "width": "30%",
          }
        }
      />
      <a
        target="_blank"
      >
        <h1
          style={
            Object {
              "color": "white",
              "paddingTop": 100,
              "textDecoration": "underline",
            }
          }
        />
      </a>
      <br />
      <p />
      <br />
      <p>
        Genre: 
      </p>
      <p>
        Episodes: 
      </p>
      <p>
        Seasons: 
      </p>
      <p>
        Rating: 
      </p>
      <p>
        Status: 
      </p>
    </div>
  </Dialog>
</div>
`;
